gensim/models/doc2vec.py

```diff
@@ -635,8 +635,13 @@ def scan_vocab(self, documents, progress_per=10000, trim_rule=None):
         min_reduce = 1
         interval_start = default_timer() - 0.00001  # guard against next sample being identical
         interval_count = 0
+        checked_string_types = 0
         vocab = defaultdict(int)
         for document_no, document in enumerate(documents):
+            if not checked_string_types:
```

Comment:
On second thoughts, we definitely want these warnings to appear in the log, so we have to check anyway (and don't log if already logged before).